### ISSUE TITLE
False positive on CA1602 when using validation method from other modules to do the null check.

### ISSUE THREAD

**Issue by r12f** [dotnet/roslyn-analyzers#2863]
#### Analyzer package
[Microsoft.CodeAnalysis.FxCopAnalyzers](https://www.nuget.org/packages/Microsoft.CodeAnalysis.FxCopAnalyzers)

#### Package Version
v2.9.5-beta1.final (Latest)

#### Diagnostic ID

Example: [CA1602](https://docs.microsoft.com/en-us/visualstudio/code-quality/ca1062-validate-arguments-of-public-methods)

#### Repro steps

Create a console app and use the following code the create the CA1602 issue, if the ArgumentNotNull check is not called.
```
using System;
using Utils;

namespace ConsoleApp
{
    class Program
    {
        static void Main()
        {
            SomeClass someClass = new SomeClass(null);
        }
    }

    public class SomeClass
    {
        public SomeClass(string s)
        {
            // RequiresInModule.ArgumentNotNull(s, nameof(s)); // No CA1602 error.
            Requires.ArgumentNotNull(s, nameof(s)); // CA1602 will still happen.
            DoSomething(s.Length);
        }

        public static void DoSomething(int i)
        {
            Console.WriteLine(i);
        }
    }

    public static class RequiresInModule
    {
        public static void ArgumentNotNull(object arg, string argName)
        {
            if (arg == null)
            {
                throw new ArgumentNullException(argName);
            }
        }
    }
}
```

Create another dll, say Utils.dll, and add the Requires class to do the null check as below.
```
using System;

namespace Utils
{
    public static class Requires
    {
        public static void ArgumentNotNull(object arg, string argName)
        {
            if (arg == null)
            {
                throw new ArgumentNullException(argName);
            }
        }
    }
}
```

#### Expected behavior
No matter which one we call to check the null, Requires from Utils.dll or RequiresInModule in the same ConsoleApp.exe, we should not see the CA1602 error.

#### Actual behavior
FxCop will still fire CA1602 error, if the null check in Utils.dll is called.

![image](https://user-images.githubusercontent.com/1533278/65488908-e57d5880-de5e-11e9-8c9c-8d62b4b67bd8.png)

**Comment by mavasani** [comment]
This is a duplicate of https://github.com/dotnet/roslyn-analyzers/issues/2525. You can use the ValidatedNotNullAttribute on the parameter. Alternatively, you can also use the following option to explicitly specify your null check validation methods in metadata: https://github.com/dotnet/roslyn-analyzers/blob/master/docs/Analyzer%20Configuration.md#null-check-validation-methods

**Comment by r12f** [comment]
Thank you so much! Somehow I only found the one talking about generic null check doesn’t work, but not this one.....
